### ISSUE TITLE
Bookmark files not found

### ISSUE THREAD

**Issue by zygous** [sciancio/searchfirefoxbookmarks#2]
When starting or restarting Gnome Shell I see a notification: "It seems are no files in /home/richardt/.mozilla/firefox/24iy4qhy.default/bookmarkbackups". This is not true though:

richardt@lynher bookmarkbackups$ pwd
/home/richardt/.mozilla/firefox/24iy4qhy.default/bookmarkbackups
richardt@lynher bookmarkbackups$ ls -la
total 284
drwxr-xr-x.  2 richardt richardt  4096 May 22 16:32 .
drwxr-xr-x. 14 richardt richardt  4096 May 23 10:04 ..
-rw-------   1 richardt richardt 24290 May  9 17:05 bookmarks-2012-05-09.json
-rw-------   1 richardt richardt 24743 May 10 10:48 bookmarks-2012-05-10.json
-rw-------   1 richardt richardt 24960 May 11 13:46 bookmarks-2012-05-11.json
-rw-------   1 richardt richardt 24960 May 14 10:34 bookmarks-2012-05-14.json
-rw-------   1 richardt richardt 24986 May 15 09:49 bookmarks-2012-05-15.json
-rw-------   1 richardt richardt 26224 May 16 17:05 bookmarks-2012-05-16.json
-rw-------   1 richardt richardt 26717 May 17 17:08 bookmarks-2012-05-17.json
-rw-------   1 richardt richardt 26717 May 18 09:22 bookmarks-2012-05-18.json
-rw-------   1 richardt richardt 26717 May 21 17:15 bookmarks-2012-05-21.json
-rw-------   1 richardt richardt 26717 May 22 16:32 bookmarks-2012-05-22.json

**Comment by sciancio** [comment]
Hi Richard,

I need some more information to undestand the problem.

Can you replace __getBookmarkFilePath_ function in 
$HOME/.local/share/gnome-shell/extensions/searchfirefoxbookmarks@ciancio.net/extension.js 
with the following:

```
 _getBookmarkFilePath : function (bookmarkDir) {

     let dir = '';
     if ((bookmarkDir) && GLib.file_test(bookmarkDir, 
```

GLib.FileTest.IS_DIR) ) {
             dir = Gio.file_new_for_path(bookmarkDir);
             var backupEnum = 
dir.enumerate_children('standard::name,standard::type,time::modified',

Gio.FileQueryInfoFlags.NONE, null);
         } else {
             Main.notifyError("Directory Error", bookmarkDir + " seems 
doesn't exist");
             return false;
         }

```
     global.log("SFB --------------------------- START");
     global.log("SFB - Dir: " + bookmarkDir);

     let infoTimeVal = new GLib.TimeVal();
     let max = 0;
     let info;
     while ((info = backupEnum.next_file(null)) != null) {

         global.log("SFB - Info: " + info.get_name());

         let type = info.get_file_type();

         if (type == Gio.FileType.REGULAR) {

             let infoTimeVal = info.get_modification_time();

             global.log("SFB - infoTimeVal.tv_sec > max: " + 
```

infoTimeVal.tv_sec + " > " + max);

```
             if (infoTimeVal.tv_sec > max) {
                 max = infoTimeVal.tv_sec;
                 var lastFile = info;

                 global.log("SFB - lastFile: " + lastFile.get_name());

             }
         }
     }
     backupEnum.close(null);

     global.log("SFB - selected file: " + lastFile.get_name());
     global.log("SFB --------------------------- END");

     if ( (typeof(lastFile) == 'undefined') ||
             !GLib.file_test(GLib.build_filenamev([bookmarkDir, 
```

lastFile.get_name()]), GLib.FileTest.EXISTS) ) {
             Main.notifyError("Directory Error", "It seems are no files 
in " + bookmarkDir);
             return false;
         }

```
     return GLib.build_filenamev([bookmarkDir, lastFile.get_name()]);
 },
```

I have added some debug output.

After replace, restart shell (Alt+F2 and enter "r" command) and view 
glass output (Alt+F2 and enter "lg" command, "Errors" tab).
I need the output (screenshot?) enclosed in "--------------------------" .

Thanks,
    Stefano

**Comment by zygous** [comment]
Well, I don't know what the issue was, but it seems to be resolved now!

I hadn't noticed this morning that I had no error when I logged-in, in fact I didn't notice until I applied your patch and saw in the Looking Glass log that the extension is now finding a bookmarks file. I don't know why rebooting the machine should make a difference when restarting Shell didn't, but there it is. Sorry about the noise.
